### DISCUSSION TITLE
is using diesel cli with planetscale possible?

### DISCUSSION THREAD

**Discussion by samifouad** [diesel-rs/diesel#3511] [Q&A]
Hey,

I just grabbed the latest versions of diesel/diesel cli and tried connecting to a planetscale-hosted mysql database. it's not working at all I'm not sure if it's even something I can control/change. planetscale says the error message is caused by mysql client attempting to modify schema. planetscale free tier doesn't allow for schema changes on the "production" schema, only on a "development" one i have setup. but i see no obvious way to configure diesel to work with my development branch to avoid the error.

planetscale has excellent support for prisma, which ive used in the past for ORM in javascript world. so it seems like something that should be possible with diesel cli, i just don't know what steps to take.

does anyone have any advice/recommendations?

**Comment by samifouad** [comment]
i was eventually able to sort out for myself by switching to an ORM (prisma) that has documentation on working with planetscale's non-traditional approach to mysql schema changes. i've still yet to test if any of what solved my problem applies to diesel, but i will come back to this sometime in the future. i've already lost way too much time today fighting something that should have been really easy.

**Comment by weiznich** [comment]
Diesel officially supports PostgreSQL, SQLite and MySQL. Any other database system is not supported by the main crate as we just do not have the maintainers that write, test and use other database systems. If you are interested in support for different database systems the recommend way is to implement a custom backend + connection as third party crate as this is for example done by `diesel-oci` or `rsfbclient-diesel`.
